ui/src/App.tsx

```diff
@@ -361,21 +361,20 @@ export default function App() {
     ) as HTMLInputElement;
     const fileInputListener = async () => {
       if (fileInput && fileInput.files && fileInput.files.length > 0) {
-        const flow = await createFlow({
-          // @ts-ignore
+        const flow = await workflowsTable?.createFlow({
           name: fileInput.files[0].name,
           json: JSON.stringify(defaultGraph),
         });
 
-        setCurFlowIDAndName(flow.id, flow.name ?? "Unknown name");
+        flow && setCurFlowIDAndName(flow.id, flow.name ?? "Unknown name");
       }
     };
     fileInput?.addEventListener("change", fileInputListener);
 
-    const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+    const handleBeforeUnload = async (e: BeforeUnloadEvent) => {
       const autoSaveEnabled = userSettingsTable?.getSetting("autoSave") ?? true;
-
-      if (!autoSaveEnabled && checkIsDirty()) {
+      const isDirty = await checkIsDirty();
+      if (!autoSaveEnabled && isDirty) {
```

Comment:
oh actually a small mistake from me, i should only add this event listener if its NOT auto save enabled
    window.addEventListener("beforeunload", handleBeforeUnload);